packages/anoncreds/src/repository/AnonCredsCredentialRecord.ts

```diff
@@ -0,0 +1,74 @@
+import type { AnonCredsCredential } from '../models'
+
+import { BaseRecord, utils } from '@aries-framework/core'
+
+export interface AnonCredsCredentialRecordProps {
+  id?: string
+  credential: AnonCredsCredential
+  credentialId: string
+  credentialRevocationId?: string
+  linkSecretId: string
+  schemaName: string
+  schemaVersion: string
+  schemaIssuerId: string
+  issuerId: string
+}
+
+export type DefaultAnonCredsCredentialTags = {
+  credentialId: string
+  linkSecretId: string
+  credentialDefinitionId: string
+  credentialRevocationId?: string
+  schemaId: string
+  attributes: string[]
+}
+
+export type CustomAnonCredsCredentialTags = {
+  schemaName: string
+  schemaVersion: string
+  schemaIssuerId: string
+  issuerId: string
+}
+
+export class AnonCredsCredentialRecord extends BaseRecord<
+  DefaultAnonCredsCredentialTags,
+  CustomAnonCredsCredentialTags
+> {
+  public static readonly type = 'AnonCredsCredentialRecord'
+  public readonly type = AnonCredsCredentialRecord.type
+
+  public readonly credentialId!: string
+  public readonly credentialRevocationId?: string
+  public readonly linkSecretId!: string
+  public readonly credential!: AnonCredsCredential
+
+  public constructor(props: AnonCredsCredentialRecordProps) {
+    super()
+
+    if (props) {
+      this.id = props.id ?? utils.uuid()
+      this.credentialId = props.credentialId
+      this.credential = props.credential
+      this.credentialRevocationId = props.credentialRevocationId
+      this.linkSecretId = props.linkSecretId
+      this.setTags({
+        issuerId: props.issuerId,
+        schemaIssuerId: props.schemaIssuerId,
+        schemaName: props.schemaName,
+        schemaVersion: props.schemaVersion,
+      })
+    }
+  }
+
+  public getTags() {
+    return {
+      ...this._tags,
+      credentialDefinitionId: this.credential.cred_def_id,
+      schemaId: this.credential.schema_id,
+      credentialId: this.credentialId,
+      credentialRevocationId: this.credentialRevocationId,
+      linkSecretId: this.linkSecretId,
+      attributes: Object.keys(this.credential.values),
```

Comment:
This will only store the credential attribute names right? I think to support all query types we have to store both the attribute name (so we can check if it exists), but also the value.

We can also do that in post filtering, as we do with JSON-LD creds, but it's a bit less efficient

Reply:
Yes, I kept only the names for the moment. I'm a bit reluctant on storing values as tags because I feel it's very arbitrary in terms of content length and amount, and with current implementation we are restricted to a single tag containing strings that are supposed to be short.

Also I don't imagine a case where a holder has hundreds or thousands credentials matching a query, that usually will include not only attribute names but also issuer/credentialDefinition/schema Ids.  Do you think it could become common?

In regards to this topic, note that `AnonCredsRsHolderService.getCredentialsForProofRequest` currently is not taking into account  predicates p_type and p_value (e.g. for a given predicate, it will return all credentials that matches its name and restrictions).  Is this OK (meaning that this will be checked in another layer) or should it manually process all predicates and only return those matching the logical conditions?

Reply:
Update: I see I missed the `attr::<attribute-name>::marker` and `attr::<attribute-name>::<attribute-value>` from the spec. So we should either:
- add as many tags as attributes the credential has, or 
- filter it in AnonCredsRsHolderService after doing the findByQuery